src/peer_digest.cc

```diff
@@ -699,7 +650,7 @@ peerDigestFetchedEnough(DigestFetchState * fetch, char *buf, ssize_t size, const
     const char *reason = nullptr;  /* reason for completion */
     const char *no_bug = nullptr;  /* successful completion if set */
     const int pdcb_valid = cbdataReferenceValid(fetch->pd);
-    const int pcb_valid = cbdataReferenceValid(fetch->pd->peer);
+    const int pcb_valid = pdcb_valid && fetch->pd->peer.valid();
```

Comment:
This is essentially a bug fix: The official code could dereference an invalid `fetch->pd` pointer here. I added the corresponding sentence to the PR description.